src/compiler/checker.ts

```diff
@@ -27150,6 +27149,11 @@ namespace ts {
             (getCheckFlags(prop) & CheckFlags.Instantiated ? getSymbolLinks(prop).target : prop)!.isReferenced = SymbolFlags.All;
         }
 
+        function isSelfTypeAccess(name: Expression | QualifiedName, parent: Symbol | undefined) {
+            return name.kind === SyntaxKind.ThisKeyword
+                || !!parent && parent === getNodeLinks(name).resolvedSymbol;

```

Comment:
I know it's used all over the place, but `getNodeLinks(name).resolvedSymbol` feels less safe than calling `getResolvedSymbol`, just in case `NodeLinks.resolvedSymbol` hasn't been set yet. Can you use that instead?